react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -8,19 +9,47 @@ const initialState = {
 
 function reducer(state = initialState, action) {
   switch (action.type) {
-    case actions.GET_BOOKS: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.REMOVE_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+    case actions.GET_BOOKS:
+      return {
+        ...state,
+        books: state.books.concat(DATA),
```

Comment:
getBooks action already have DATA as a paylod, so you can only write books: action.payload and so on